### PR TITLE
Fix token client functions, simple-nft in Python SDK

### PR DESCRIPTION
**Pull request by banool**
### Description
- Make the token functions in the Python SDK work the same as the TS SDK, namely return something on 404 instead of throwing.
- Change the Python SDK to raise an exception instead of assert when response >= 400.
- Format the examples as well.
- Fix tab stuff in the first NFT example.

I have made sure the docs still look good since the section tags moved around.

### Test Plan
```
cd ecosystem/python/sdk
curl -sSL https://install.python-poetry.org/ | python3
poetry update
poetry run python -m examples.simple-nft
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3710)

### CODE REVIEW

File: developer-docs-site/docs/tutorials/first-dapp.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ For this tutorial, we will use the Move Module `HelloBlockchain` described in [Y
 
 We will use:
 
-- The [Aptos SDK](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk).
-- The [Aptos Wallet](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/web-wallet), and
-- The [Aptos CLI](https://github.com/aptos-labs/aptos-core/tree/main/crates/aptos) to interact with blockchain.
+- The [Aptos Typescript SDK][ts_sdk],
```
Comment (areshand) [inline developer-docs-site/docs/tutorials/first-dapp.md:14]: do we still have links?

File: ecosystem/python/sdk/aptos_sdk/client.py [hunk 2]
```diff
@@ -458,24 +463,47 @@ def get_token_balance(
...
 
-        #:!:>read_token_data_table
+    def get_token_balance(
+        self,
+        owner: AccountAddress,
+        creator: AccountAddress,
+        collection_name: str,
+        token_name: str,
+        property_version: int,
+    ) -> Any:
```
[the remark this answers is not in the document]
Comment (banool) [reply ecosystem/python/sdk/aptos_sdk/client.py:489]: Will change to str to match the other change.

File: developer-docs-site/docs/tutorials/first-dapp.md (hunk 1 above)
Comment (banool) [reply developer-docs-site/docs/tutorials/first-dapp.md:14]: Yeah this is a funky docusaurus syntax, see the bottom of the file.

File: ecosystem/python/sdk/aptos_sdk/client.py [hunk 3]
```diff
@@ -458,24 +463,47 @@ def get_token_balance(
...
         collection_name: str,
         token_name: str,
         property_version: int,
     ) -> Any:
-        token_data = self.account_resource(creator, "0x3::token::Collections")["data"][
-            "token_data"
-        ]["handle"]
+        token_data_handle = self.account_resource(creator, "0x3::token::Collections")[
+            "data"
+        ]["token_data"]["handle"]
```
[the remark this answers is not in the document]
Comment (banool) [reply ecosystem/python/sdk/aptos_sdk/client.py:506]: This is just copy pasta, we can clean this up wholesale later.

**Review by gregnazario — APPROVED** [review]

File: ecosystem/python/sdk/aptos_sdk/client.py [hunk 4]
```diff
@@ -311,8 +317,7 @@ def create_token(
         supply: int,
         uri: str,
         royalty_points_per_million: int,
-    ) -> str:
-#<:!:create_token
+    ) -> str:  # <:!:create_token
```
[the remark this answers is not in the document]
Comment (banool) [reply ecosystem/python/sdk/aptos_sdk/client.py:320]: Nup

**Pull request merged**